tests/rptest/services/redpanda_installer.py

```diff
@@ -341,25 +352,79 @@ def highest_from_prior_feature_version(self, version):
         )
         return result
 
-    def install(self, nodes, version):
+    def latest_for_line(self, release_line: tuple[int, int]):
+        """
+        Returns the most recent version of redpanda from a release line, or HEAD if asking for a yet-to-be released version
+        the return type is a tuple (version, is_head), where is_head is True if the version is from dev tip
+        e.g: latest_for_line((22, 2)) -> ((22, 2, 7), False)
+        latest_for_line((23, 1)) -> (self._head_version, True) (as of 2022 dec (23, 1, 0))
+        """
+        # NOTE: _released_versions are in descending order.
+
+        self.start()
+        self._initialize_released_versions()
+
+        # if requesting current (or future) release line, return _head_version
+        if release_line >= self._head_version[0:2]:
+            self._redpanda.logger.info(
+                f"selecting HEAD={self._head_version} for {release_line=}")
+            return (self._head_version, True)
+
+        line = [v for v in self._released_versions if release_line == v[0:2]]
+        assert len(
+            line
+        ) > 0, f"could not find a line for {release_line=} in {self._released_versions=}"
+
+        # Only checks these many version before giving up. one missing version is fine in a transient state,
+        # but more would indicate a systemic issues in package download
+        for v in line[0:2]:
+            # check actual availability
+            if self._avail_for_download(v):
+                self._redpanda.logger.info(
+                    f"selecting {v=} for {release_line=}")
+                return (v, False)
```

Comment:
nit: could you add a warning log that we're skipping a version because it's unavailable? It'll be obvious for those who know how the installer work, but could be otherwise surprising just seeing a failed test behaving a certain way because of a cloudsmith issue